--- amanda-2.4.2p2/tape-src/Makefile.am~	Sat Feb  3 23:17:11 2001
+++ amanda-2.4.2p2/tape-src/Makefile.am	Fri Jun  1 14:59:50 2001
@@ -29,9 +29,9 @@
 # routines, and second to pick up any references in the other libraries.
 ###
 
-tapetype_LDADD =	../common-src/libamanda.$(LIB_EXTENSION) \
-			libamtape.$(LIB_EXTENSION) \
-			../common-src/libamanda.$(LIB_EXTENSION)
+tapetype_LDADD =	../common-src/libamanda.la \
+			libamtape.la \
+			../common-src/libamanda.la
 
 ammt_LDADD =		../common-src/libamanda.$(LIB_EXTENSION) \
 			libamtape.$(LIB_EXTENSION) \
@@ -49,7 +49,7 @@
 
 TEST_PROGS =		amtapeio
 
-EXTRA_PROGRAMS = 	tapetype $(TEST_PROGS)
+sbin_PROGRAMS = 	tapetype $(TEST_PROGS)
 
 CLEANFILES = *.test.c
 
