--- amanda-2.4.2p2/configure.in.orig	Tue Aug 27 09:17:09 2002
+++ amanda-2.4.2p2/configure.in	Tue Aug 27 09:22:07 2002
@@ -2265,7 +2265,7 @@
 ICE_CHECK_DECL(vsnprintf,stdio.h)
 if test x"$ice_have_snprintf" != x"yes" ||
    test x"$ice_have_vsnprintf" != x"yes"; then
-    LIBOBJS="$LIBOBJS snprintf.${ac_objext}"
+    AC_LIBOBJ(snprintf)
     if false; then :
 	dnl so that automake includes snprintf.c in the distribution
 	AC_REPLACE_FUNCS(snprintf) 
@@ -2513,7 +2513,7 @@
 
 AC_SUBST(LTLIBOBJS)
 AC_SUBST(LTALLOCA)
-LTLIBOBJS=`echo "$LIBOBJS" | sed 's/\.'"${ac_objext}"'/\.lo/g'`
+LTLIBOBJS=`echo "$LIB@&t@OBJS" | sed 's/\.'"${ac_objext}"'/\.lo/g'`
 LTALLOCA=`echo "$ALLOCA" | sed 's/\.'"${ac_objext}"'/\.lo/g'`
 
 AC_OUTPUT( \
