diff -Nru amanda-2.4.1p1/amplot/Makefile.am amanda-2.4.1p1.new/amplot/Makefile.am
--- amanda-2.4.1p1/amplot/Makefile.am	Mon Nov 29 22:15:02 1999
+++ amanda-2.4.1p1.new/amplot/Makefile.am	Mon Nov 29 22:16:10 1999
@@ -15,8 +15,7 @@
 amplotdir =		$(libexecdir)
 amplot_DATA =		@BUILD_AMPLOT_LIBEXEC@
 
-EXTRA_DATA =		amplot.awk		amplot.g
-EXTRA_DIST =		$(EXTRA_DATA)
+EXTRA_DIST =		amplot.awk		amplot.g
 
 CLEANFILES = amplot.gp.new
 DISTCLEANFILES = amplot.gp
@@ -30,7 +29,7 @@
 install-exec-hook:
 	@list="$(sbin_SCRIPTS)"; \
 	for p in $$list; do \
-		pa=$(sbindir)/`echo $$p|sed '$(transform)'`; \
+		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
 		echo chown $(BINARY_OWNER) $$pa; \
 		chown $(BINARY_OWNER) $$pa; \
 		echo chgrp $(SETUID_GROUP) $$pa; \
@@ -40,7 +39,7 @@
 install-data-hook:
 	@list="$(amplot_DATA)"; \
 	for p in $$list; do \
-		pa=$(libexecdir)/$$p; \
+		pa=$(DESTDIR)$(libexecdir)/$$p; \
 		echo chown $(BINARY_OWNER) $$pa; \
 		chown $(BINARY_OWNER) $$pa; \
 		echo chgrp $(SETUID_GROUP) $$pa; \
