diff -Nru amanda-2.4.2/Makefile.am amanda-2.4.2.new/Makefile.am
--- amanda-2.4.2/Makefile.am	Wed Nov 22 14:39:10 2000
+++ amanda-2.4.2.new/Makefile.am	Fri Jan 19 07:07:44 2001
@@ -36,13 +36,10 @@
 	$(PLOT_SUBDIRS) \
 	man docs example
 
-pkgdata_DATA = \
+EXTRA_DIST = $(SNAPSHOT_STAMP) \
 	COPYRIGHT			\
 	COPYRIGHT-APACHE		\
-	COPYRIGHT-REGEX
-
-EXTRA_DIST = $(SNAPSHOT_STAMP) \
-	$(pkgdata_DATA)			\
+	COPYRIGHT-REGEX			\
 	contrib/README			\
 	contrib/dbbackup.README		\
 	contrib/dbbackup.ksh		\
diff -Nru amanda-2.4.2/amplot/Makefile.am amanda-2.4.2.new/amplot/Makefile.am
--- amanda-2.4.2/amplot/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/amplot/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -22,23 +22,3 @@
 			echo "pause -1;" >> amplot.gp.new
 			rm -f amplot.gp
 			mv amplot.gp.new amplot.gp
-
-install-exec-hook:
-	@list="$(sbin_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
-
-install-data-hook:
-	@list="$(amplot_DATA)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(libexecdir)/$$p; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-        done
diff -Nru amanda-2.4.2/changer-src/Makefile.am amanda-2.4.2.new/changer-src/Makefile.am
--- amanda-2.4.2/changer-src/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/changer-src/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -41,21 +41,3 @@
 .sh:
 			cat $< > $@
 			chmod a+x $@
-
-install-exec-hook:
-	@list="$(sbin_PROGRAMS) $(sbin_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
-	@list="$(libexec_PROGRAMS) $(libexec_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(libexecdir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
diff -Nru amanda-2.4.2/client-src/Makefile.am amanda-2.4.2.new/client-src/Makefile.am
--- amanda-2.4.2/client-src/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/client-src/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -57,22 +57,6 @@
 noinst_HEADERS	= 	amandates.h	getfsent.h	findpass.h
 
 install-exec-hook:
-	@list="$(sbin_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
-	@list="$(libexec_PROGRAMS) $(libexec_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(libexecdir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
 	@list="$(libexecdir)/calcsize $(libexecdir)/killpgrp $(libexecdir)/rundump $(libexecdir)/runtar"; \
 	for p in $$list; do \
 		if echo "$(sbin_SCRIPTS) $(libexec_PROGRAMS) $(libexec_SCRIPTS)" | grep `basename $$p` >/dev/null 2>&1; then \
diff -Nru amanda-2.4.2/docs/Makefile.am amanda-2.4.2.new/docs/Makefile.am
--- amanda-2.4.2/docs/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/docs/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -1,5 +1,5 @@
 ## Process this file with automake to produce Makefile.in
-pkgdata_DATA = \
+EXTRA_DIST = \
 	DUMPER-API		\
 	FAQ			\
 	INDEXING		\
diff -Nru amanda-2.4.2/man/Makefile.am amanda-2.4.2.new/man/Makefile.am
--- amanda-2.4.2/man/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/man/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -42,13 +42,3 @@
 	   $(SERVER_MAN_PAGES) \
 	   $(RECOVER_MAN_PAGES) \
 	   $(RESTORE_MAN_PAGES)
-
-install-data-hook:
-	@list="$(man_MANS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(mandir)/man8/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
diff -Nru amanda-2.4.2/recover-src/Makefile.am amanda-2.4.2.new/recover-src/Makefile.am
--- amanda-2.4.2/recover-src/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/recover-src/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -24,18 +24,6 @@
 
 YFLAGS =		-d
 
-install-exec-hook:
-	@list="$(sbin_PROGRAMS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-		echo chmod o-rwx $$pa; \
-		chmod o-rwx $$pa; \
-	done
-
 # so that uscan.c is never generated before uparse.h
 # otherwise we might have makedepend problems
 $(srcdir)/uscan.c: $(srcdir)/uparse.h
diff -Nru amanda-2.4.2/restore-src/Makefile.am amanda-2.4.2.new/restore-src/Makefile.am
--- amanda-2.4.2/restore-src/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/restore-src/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -16,21 +16,3 @@
 	../common-src/libamanda.$(LIB_EXTENSION)
 
 amidxtaped_SOURCES = 	amidxtaped.c
-
-install-exec-hook:
-	@list="$(sbin_PROGRAMS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
-	@list="$(libexec_PROGRAMS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(libexecdir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
diff -Nru amanda-2.4.2/server-src/Makefile.am amanda-2.4.2.new/server-src/Makefile.am
--- amanda-2.4.2/server-src/Makefile.am	Fri Jan 19 07:07:01 2001
+++ amanda-2.4.2.new/server-src/Makefile.am	Fri Jan 19 07:06:35 2001
@@ -64,22 +64,6 @@
 			tapefile.h	find.h		server_util.h
 
 install-exec-hook:
-	@list="$(sbin_PROGRAMS) $(sbin_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(sbindir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
-	@list="$(libexec_PROGRAMS) $(libexec_SCRIPTS)"; \
-	for p in $$list; do \
-		pa=$(DESTDIR)$(libexecdir)/`echo $$p|sed '$(transform)'`; \
-		echo chown $(BINARY_OWNER) $$pa; \
-		chown $(BINARY_OWNER) $$pa; \
-		echo chgrp $(SETUID_GROUP) $$pa; \
-		chgrp $(SETUID_GROUP) $$pa; \
-	done
 	@list="$(sbindir)/amcheck $(libexecdir)/dumper $(libexecdir)/planner"; \
 	for p in $$list; do \
 		if echo "$(sbin_PROGRAMS) $(sbin_SCRIPTS) $(libexec_PROGRAMS) $(libexec_SCRIPTS)" | grep `basename $$p` >/dev/null 2>&1; then \
--- amanda-2.4.2p2/docs/Makefile.am~	Wed Apr 18 11:54:43 2001
+++ amanda-2.4.2p2/docs/Makefile.am	Wed Apr 18 11:58:02 2001
@@ -19,5 +19,3 @@
 	WISHLIST		\
 	YEAR2000		\
 	ZFTAPE
-
-EXTRA_DIST = $(pkgdata_DATA)
