--- amanda-2.4.2p2/configure.in.orig	Wed Sep 19 09:50:57 2001
+++ amanda-2.4.2p2/configure.in	Wed Sep 19 10:06:27 2001
@@ -1441,7 +1441,7 @@
 if test "$XFSDUMP" -a "$XFSRESTORE"; then
     AC_DEFINE_UNQUOTED(XFSDUMP,"$XFSDUMP",[Define the location of the xfsdump program on Irix hosts. ])
     AC_DEFINE_UNQUOTED(XFSRESTORE,"$XFSRESTORE",[Define the location of the xfsrestore program on Irix hosts. ])
-    AC_MSG_WARN([*** xfsdump causes the setuid-root rundump program to be enabled])
+    AC_MSG_WARN([[*** xfsdump causes the setuid-root rundump program to be enabled]])
     AC_MSG_WARN([[*** to disable it, just #undef XFSDUMP in config/config.h]])
 fi
 
