--- amanda-3.5.1/man/Makefile.am~	2017-12-01 14:26:28.000000000 +0100
+++ amanda-3.5.1/man/Makefile.am	2018-09-20 08:56:14.538624065 +0200
@@ -19,6 +19,11 @@ COMMON_MAN_PAGES =  amanda.8 \
 		    amgetconf.8 \
 		    amservice.8 \
 		    amssl.8 \
+		    amcrypt-ossl-asym.8 \
+		    amcrypt-ossl.8   \
+		    amcrypt.8 \
+		    amcryptsimple.8  \
+		    amgpgcrypt.8 \
 		    script-email.8
 
 CLIENT_MAN_PAGES = \
@@ -49,14 +54,9 @@ SERVER_MAN_PAGES =  \
     amcheckdump.8 \
     amcleanup.8 \
     amcleanupdisk.8 \
-    amcrypt-ossl-asym.8 \
-    amcrypt-ossl.8   \
-    amcrypt.8 \
-    amcryptsimple.8  \
     amdevcheck.8 \
     amdump.8 \
     amflush.8 \
-    amgpgcrypt.8 \
     amlabel.8 \
     amoverview.8 \
     amreindex.8 \
