diff -Nur amanda-2.4.2p2.orig/client-src/Makefile.am amanda-2.4.2p2/client-src/Makefile.am
--- amanda-2.4.2p2.orig/client-src/Makefile.am	Tue Aug 27 09:25:46 2002
+++ amanda-2.4.2p2/client-src/Makefile.am	Tue Aug 27 09:37:09 2002
@@ -2,13 +2,8 @@
 
 INCLUDES =		-I$(top_srcdir)/common-src
 
-if WANT_LIBTOOL
 lib_LTLIBRARIES =	libamclient.la
 LIB_EXTENSION = la
-else
-noinst_LIBRARIES =	libamclient.a
-LIB_EXTENSION = a
-endif
 
 libexec_PROGRAMS =	amandad calcsize killpgrp rundump runtar selfcheck sendbackup sendsize versionsuffix
 
@@ -25,8 +20,6 @@
 
 libamclient_la_LDFLAGS = -release $(VERSION)
 
-libamclient_a_SOURCES=  $(libamclient_la_SOURCES)
-
 ###
 # Because libamanda includes routines (e.g. regex) provided by some system
 # libraries, and because of the way libtool sets up the command line, we
diff -Nur amanda-2.4.2p2.orig/common-src/Makefile.am amanda-2.4.2p2/common-src/Makefile.am
--- amanda-2.4.2p2.orig/common-src/Makefile.am	Sat Mar 17 03:56:56 2001
+++ amanda-2.4.2p2/common-src/Makefile.am	Tue Aug 27 09:39:10 2002
@@ -1,10 +1,6 @@
 # Makefile for Amanda library.
 
-if WANT_LIBTOOL
 lib_LTLIBRARIES =	libamanda.la
-else
-noinst_LIBRARIES = 	libamanda.a
-endif
 
 REGDIR = ../regex-src
 REGsrcdir = $(srcdir)/$(REGDIR)
@@ -22,8 +18,7 @@
 libamanda_la_LIBADD =	@LTLIBOBJS@ @LTALLOCA@
 libamanda_la_LDFLAGS =  -release $(VERSION)
 
-libamanda_a_SOURCES =	$(libamanda_la_SOURCES)
-libamanda_a_LIBADD =	@LIBOBJS@ @ALLOCA@
+libobj_libadd =	@LIBOBJS@ @ALLOCA@
 
 noinst_HEADERS =	amanda.h	arglist.h	\
 			dgram.h		protocol.h	\
@@ -48,8 +43,8 @@
 
 EXTRA_PROGRAMS = genversion $(TEST_PROGS)
 
-genversion_SOURCES = genversion.c versuff.c
-genversion_LDADD = $(libamanda_a_LIBADD) $(STANDARD_COMMON_STUFF)
+genversion_SOURCES = genversion.c
+genversion_LDADD = $(libobj_libadd) $(STANDARD_COMMON_STUFF) versuff.lo
 
 genversion.o: genversion.h
 genversion.h: $(top_builddir)/config.status
@@ -97,16 +92,16 @@
 # used for testing only
 
 statfs_SOURCES = statfs.test.c
-statfs_LDADD = $(libamanda_a_LIBADD) $(STANDARD_COMMON_STUFF)
+statfs_LDADD = $(libobj_libadd) $(STANDARD_COMMON_STUFF)
 
 token_SOURCES = token.test.c
-token_LDADD = $(libamanda_a_LIBADD) $(STANDARD_COMMON_STUFF)
+token_LDADD = $(libobj_libadd) $(STANDARD_COMMON_STUFF)
 
 file_SOURCES = file.test.c
-file_LDADD = $(libamanda_a_LIBADD) $(STANDARD_COMMON_STUFF_NOT_FILE)
+file_LDADD = $(libobj_libadd) $(STANDARD_COMMON_STUFF_NOT_FILE)
 
 security_SOURCES = security.test.c
-security_LDADD = $(libamanda_a_LIBADD) $(STANDARD_COMMON_STUFF)
+security_LDADD = $(libobj_libadd) $(STANDARD_COMMON_STUFF)
 
 %.test.c: $(srcdir)/%.c
 	echo '#define TEST' >$@
diff -Nur amanda-2.4.2p2.orig/server-src/Makefile.am amanda-2.4.2p2/server-src/Makefile.am
--- amanda-2.4.2p2.orig/server-src/Makefile.am	Tue Aug 27 09:25:46 2002
+++ amanda-2.4.2p2/server-src/Makefile.am	Tue Aug 27 09:37:09 2002
@@ -2,13 +2,8 @@
 
 INCLUDES =		-I$(top_srcdir)/common-src -I$(top_srcdir)/tape-src
 
-if WANT_LIBTOOL
 lib_LTLIBRARIES = 	libamserver.la
 LIB_EXTENSION = la
-else
-noinst_LIBRARIES = 	libamserver.a
-LIB_EXTENSION = a
-endif
 
 sbin_PROGRAMS =		amadmin		amcheck		amflush		\
 			amgetconf	amlabel		amtape		\
@@ -29,8 +24,6 @@
 
 libamserver_la_LDFLAGS= -release $(VERSION)
 
-libamserver_a_SOURCES=	$(libamserver_la_SOURCES)
-
 ###
 # Because libamanda includes routines (e.g. regex) provided by some system
 # libraries, and because of the way libtool sets up the command line, we
diff -Nur amanda-2.4.2p2.orig/tape-src/Makefile.am amanda-2.4.2p2/tape-src/Makefile.am
--- amanda-2.4.2p2.orig/tape-src/Makefile.am	Tue Aug 27 09:25:46 2002
+++ amanda-2.4.2p2/tape-src/Makefile.am	Tue Aug 27 09:37:09 2002
@@ -2,13 +2,8 @@
 
 INCLUDES =		-I$(top_srcdir)/common-src
 
-if WANT_LIBTOOL
 lib_LTLIBRARIES = 	libamtape.la
 LIB_EXTENSION = la
-else
-noinst_LIBRARIES = 	libamtape.a
-LIB_EXTENSION = a
-endif
 
 sbin_PROGRAMS=          ammt amdd
 
@@ -20,8 +15,6 @@
 
 libamtape_la_LDFLAGS =  -release $(VERSION)
 
-libamtape_a_SOURCES =	$(libamtape_la_SOURCES)
-
 ###
 # Because libamanda includes routines (e.g. regex) provided by some system
 # libraries, and because of the way libtool sets up the command line, we
