--- amanda-2.4.1p1/common-src/debug.c.tim2	Tue Jan 11 11:02:43 2000
+++ amanda-2.4.1p1/common-src/debug.c	Tue Jan 11 11:04:36 2000
@@ -42,7 +42,8 @@
   static FILE *db_file = NULL;
 #else
   int db_fd = 2;
-  static FILE *db_file = stderr;
+	/* stderr cannot be assumed to be constant */
+  static FILE *db_file = NULL;
 #endif
 
 #ifndef DEBUG_DIR
@@ -56,7 +57,10 @@
 {
     va_list argp;
 
-    if(db_fd == -1) return;
+    if(db_fd == NULL) {
+			if (db_fd == -1) return;
+			if (db_fd == 2) db_file = stderr;
+		}
 
     arglist_start(argp, format);
     vfprintf(db_file, format, argp);
