diff -urNp -x '*.orig' amanda-3.5.1.org/common-src/krb5-security.c amanda-3.5.1/common-src/krb5-security.c
--- amanda-3.5.1.org/common-src/krb5-security.c	2017-12-01 14:26:28.000000000 +0100
+++ amanda-3.5.1/common-src/krb5-security.c	2021-05-08 10:23:14.430132083 +0200
@@ -43,7 +43,7 @@
 #include "sockaddr-util.h"
 
 #ifdef KRB5_HEIMDAL_INCLUDES
-#include "com_err.h"
+#include "et/com_err.h"
 #endif
 
 #define BROKEN_MEMORY_CCACHE
@@ -65,7 +65,7 @@
 #ifndef KRB5_HEIMDAL_INCLUDES
 #include <gssapi/gssapi_generic.h>
 #else
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #endif
 #include <krb5.h>
 
diff -urNp -x '*.orig' amanda-3.5.1.org/common-src/security-util.h amanda-3.5.1/common-src/security-util.h
--- amanda-3.5.1.org/common-src/security-util.h	2017-12-01 14:26:28.000000000 +0100
+++ amanda-3.5.1/common-src/security-util.h	2021-05-08 10:23:14.430132083 +0200
@@ -53,11 +53,11 @@
 #define H_EOF   -2		/* this connection has been shut down */
 
 #ifdef KRB5_SECURITY
-#  define KRB5_DEPRECATED 1
 #  ifndef KRB5_HEIMDAL_INCLUDES
+#    define KRB5_DEPRECATED 1
 #    include <gssapi/gssapi_generic.h>
 #  else
-#    include <gssapi/gssapi.h>
+#    include <gssapi.h>
 #  endif
 #  include <krb5.h>
 #endif
diff -urNp -x '*.orig' amanda-3.5.1.org/config/amanda/krb5-security.m4 amanda-3.5.1/config/amanda/krb5-security.m4
--- amanda-3.5.1.org/config/amanda/krb5-security.m4	2017-12-01 14:26:31.000000000 +0100
+++ amanda-3.5.1/config/amanda/krb5-security.m4	2021-05-08 10:23:14.430132083 +0200
@@ -61,6 +61,8 @@ AC_DEFUN([AMANDA_KRB5_SECURITY],
                 AMANDA_ADD_LIBS([-lgssapi_krb5 -lkrb5 $K5CRYPTO $K5SUPPORT -lcom_err])
                 break
             elif test \( -f ${k5libdir}/libkrb5.a -o -f ${k5libdir}/libkrb5.so \) -a \( -f ${k5libdir}/libasn1.a -o -f ${k5libdir}/libasn1.so \) -a \( -f ${k5libdir}/libgssapi.a -o -f ${k5libdir}/libgssapi.so \); then
+                KRB5_DIR_FOUND=$dir
+                KRB5_LIBDIR_FOUND=$k5libdir
                 AMANDA_ADD_LIBS([-lgssapi -lkrb5 -lasn1])
                 AMANDA_ADD_CPPFLAGS([-DKRB5_HEIMDAL_INCLUDES])
                 break
