--- amanda-2.6.0/server-src/amtoc.pl.orig	2008-10-26 15:21:44.691836813 +0100
+++ amanda-2.6.0/server-src/amtoc.pl	2008-10-26 15:32:17.076007801 +0100
@@ -191,10 +191,14 @@
       &pr("#","Server","/partition","date", "level","size[Kb]","part");
       &pr("$filenumber","$chunk","","$disk","-","-","-");
       last switch; };
-    /^(?:SUCCESS|CHUNK) taper/ && do {
-      if(/SUCCESS/){
-	$level = $chunk;
-	$chunk = "-";
+    /^(?:PART) taper/ && do {
+      if(/PART/){
+	/^([A-Z]+) taper (\S+) (\S+) (\S+) (\S+) (\S+) (\S+) (\S+)/;
+	$host = $4;
+	$disk = $5;
+	$date = $6;
+	$chunk = $7;
+	$level = $8;
       }
       $mysize = 0;
       if(/ kb (\d+) /){
@@ -206,11 +210,11 @@
   	if (defined($flash_mode)) {
           &pr("$filenumber","${host}","${disk}","${date}","${level}","$mysize","${chunk}");
 	} else {
-  	  if (defined($osize{$host}{$disk}) && !/^CHUNK/) {
+  	  if (defined($osize{$host}{$disk}) && !/^PART/) {
             &pr("$filenumber","${host}","${disk}","${date}","${level}","$osize{$host}{$disk}","${chunk}");
 	  } else {
 	    $note = "";
-	    if(!/^CHUNK/){
+	    if(!/^PART/){
 		# this case should never happend: 
 	    $strange=1;
 	      $note = "*";
