--- amanda-3.2.3/application-src/amstar.c.orig	2012-05-09 14:36:45.583768213 +0200
+++ amanda-3.2.3/application-src/amstar.c	2012-05-10 14:41:43.738866665 +0200
@@ -454,7 +454,9 @@
     }
 
     set_root_privs(1);
-    if (argument->dle.device) {
+    if (star_directory) {
+	check_dir(star_directory, R_OK);
+    } else if (argument->dle.device) {
 	check_dir(argument->dle.device, R_OK);
     }
     set_root_privs(0);
@@ -499,9 +501,13 @@
 	fprintf(stderr, "ERROR include-list not supported for backup\n");
     }
 
-    if (check_device(argument) == 0) {
-	return;
+    set_root_privs(1);
+    if (star_directory) {
+	check_dir(star_directory, R_OK|X_OK);
+    } else if (argument->dle.device) {
+	check_dir(argument->dle.device, R_OK|X_OK);
     }
+    set_root_privs(0);
 
     qdisk = quote_string(argument->dle.disk);
     if (argument->calcsize) {
