--- amanda-2.5.2p1/configure.in~	2007-06-22 15:34:50.000000000 +0200
+++ amanda-2.5.2p1/configure.in	2007-06-22 15:38:46.000000000 +0200
@@ -851,7 +851,7 @@
 AC_MSG_CHECKING(for Kerberos and Amanda kerberos4 bits)
 if test "x${KRB4_SECURITY}" = xyes -a -f  ${srcdir-.}/common-src/krb4-security.c ; then
     for dir in $KRB4_SPOTS; do
-	if test -f ${dir}/lib/libkrb.a -a -f ${dir}/lib/libdes.a ; then
+	if test -f ${dir}/lib/libkrb.so -a -f ${dir}/lib/libdes.so ; then
 	    #
 	    # This is the original Kerberos 4.
 	    #
@@ -868,13 +868,13 @@
 	    fi
 	    KRB4LDFLAGS=-L$dir/lib
 	    KRB4LIBS="-lkrb -ldes"
-	    if test -f ${dir}/lib/libcom_err.a; then
+	    if test -f ${dir}/lib/libcom_err.so; then
 		KRB4LIBS="$KRB4LIBS -lcom_err"
 	    fi
 	    break
-	elif test -f ${dir}/lib/libkrb4.a &&
-	     test -f ${dir}/lib/libcrypto.a &&
-	     test -f ${dir}/lib/libdes425.a ; then
+	elif test -f ${dir}/lib/libkrb4.so &&
+	     test -f ${dir}/lib/libcrypto.so &&
+	     test -f ${dir}/lib/libdes425.so ; then
 	    #
 	    # This is Kerberos 5 with Kerberos 4 back-support.
 	    #
@@ -883,8 +883,8 @@
 	    AC_DEFINE(KRB4_SECURITY, 1, [Enable Kerberos security. ])
 	    KRB4INCLUDES="-I$dir/include -I$dir/include/kerberosIV"
 	    KRB4LDFLAGS=-L$dir/lib
-	    if test -f ${dir}/lib/libkrb5.a &&
-               test -f ${dir}/lib/libcom_err.a; then
+	    if test -f ${dir}/lib/libkrb5.so &&
+               test -f ${dir}/lib/libcom_err.so; then
 		KRB4LIBS="-lkrb4 -lkrb5 -lcrypto -ldes425 -lcom_err"
 	    else
 		KRB4LIBS="-lkrb4 -lcrypto -ldes425"
@@ -1135,15 +1135,15 @@
 for dir in $KRB5_SPOTS; do
   for lib in lib lib64; do
     k5libdir=${dir}/${lib}
-    if test -f ${k5libdir}/libkrb5.a -a -f ${k5libdir}/libgssapi_krb5.a -a -f ${k5libdir}/libcom_err.a; then
-	if test -f ${k5libdir}/libk5crypto.a; then
+    if test -f ${k5libdir}/libkrb5.so -a -f ${k5libdir}/libgssapi_krb5.so -a -f ${k5libdir}/libcom_err.so; then
+	if test -f ${k5libdir}/libk5crypto.so; then
 	    K5CRYPTO=-lk5crypto
-	elif test -f ${k5libdir}/libcrypto.a; then
+	elif test -f ${k5libdir}/libcrypto.so; then
 	    K5CRYPTO=-lcrypto
 	else
 	    K5CRYPTO=""
 	fi
-	if test -f ${k5libdir}/libkrb5support.a; then
+	if test -f ${k5libdir}/libkrb5support.so; then
 	    K5SUPPORT=-lkrb5support
 	else
 	    K5SUPPORT=""
@@ -1152,9 +1152,9 @@
 	KRB5LIBS="-lgssapi_krb5 -lkrb5 $K5CRYPTO $K5SUPPORT -lcom_err"
 	KRB5CFLAGS=""
 	break
-    elif test -f ${k5libdir}/libkrb5.a -a -f ${k5libdir}/libasn1.a -a -f ${k5libdir}/libgssapi.a; then
+    elif test -f ${k5libdir}/libkrb5.so -a -f ${k5libdir}/libasn1.so -a -f ${k5libdir}/libgssapi.so; then
 	KRB5_DIR_FOUND=$dir
-	KRB5LIBS="-lgssapi.a -lkrb5.a -lasn1.a"
+	KRB5LIBS="-lgssapi -lkrb5 -lasn1"
 	KRB5_CFLAGS="-DKRB5_HEIMDAL_INCLUDES"
 	break
     fi
