--- amanda/application-src/amstar.c	2010/09/24 21:46:25	3449
+++ amanda/application-src/amstar.c	2010/11/10 11:22:58	3609
@@ -993,7 +993,7 @@
 	for (excl = argument->dle.exclude_file->first; excl != NULL;
 	     excl = excl->next) {
 	    char *ex;
-	    if (strcmp(excl->name, "./") == 0) {
+	    if (strncmp(excl->name, "./", 2) == 0) {
 		ex = g_strdup_printf("pat=%s", excl->name+2);
 	    } else {
 		ex = g_strdup_printf("pat=%s", excl->name);
@@ -1013,7 +1013,7 @@
 		while ((aexc = agets(exclude)) != NULL) {
 		    if (aexc[0] != '\0') {
 			char *ex;
-			if (strcmp(aexc, "./") == 0) {
+			if (strncmp(aexc, "./", 2) == 0) {
 			    ex = g_strdup_printf("pat=%s", aexc+2);
 			} else {
 			    ex = g_strdup_printf("pat=%s", aexc);
