diff -urNp -x '*.orig' amanda-3.5.1.org/perl/Amanda/Xfer.swg amanda-3.5.1/perl/Amanda/Xfer.swg
--- amanda-3.5.1.org/perl/Amanda/Xfer.swg	2017-12-01 14:26:29.000000000 +0100
+++ amanda-3.5.1/perl/Amanda/Xfer.swg	2021-05-08 10:23:15.580125519 +0200
@@ -614,8 +614,6 @@ XFER_ELEMENT_SUBCLASS_OF(Amanda::Xfer::E
 %perlcode %{
 sub new { 
     my $pkg = shift;
-    # The C function adds the proper blessing -- this function
-    # just gets $pkg out of the way.
     C_CONSTRUCTOR(@_);
 }
 %}
@@ -624,7 +622,6 @@ sub new {
 %define OVERLOAD_REPR()
 %perlcode %{
 use overload '""' => sub { $_[0]->repr(); };
-# overload comparison, so users can ask if one obj == another
 use overload '==' => sub {     Amanda::Xfer::same_elements($_[0], $_[1]); };
 use overload '!=' => sub { not Amanda::Xfer::same_elements($_[0], $_[1]); };
 %}
